lambda/security-groups/src/main/scala/com/gu/hq/Notifier.scala

```diff
@@ -16,36 +16,43 @@ import scala.util.control.NonFatal
 object Notifier extends StrictLogging {
   private val subject = "Open Security Group Notification"
   private val sourceSystem = "Security HQ - Security Groups Lambda"
-  private val channel = HangoutsChat
+  private val channel = Preferred(HangoutsChat)  // use hangouts if available, but email is ok too
+
+  def shouldNotify(invokingEvent: InvokingEvent, status: SgStatus): Option[Unit] = {
```

Comment:
Great question, I wrestled with that.

Writing `if`s in for comps isn't great though. It means we've brought logic into the form comp (how do we test we got the if statement the right way round?). It's also an unusual use of syntax. It exists in Scala but it is one of those things you rarely see in practice (probably because of the reasons above and below), and surprising code isn't ideal.

It's also not generally true that putting an if in a for comp works. It does work for `Option` so it would be ok here and we shouldn't make speculative work, but if we wanted to carry the error with a `Try` or an `Either` (which we probably do want to do in PROD) then it would definitely make sense for this function to return the container, so it can choose the error message.

It was a close decision but these things combined made me use an Option. Let's chat IRL.